### ISSUE TITLE
Enhancement: [ban-types] Rule does not trigger with comments inside type object

### ISSUE THREAD

**Issue by yuki2006** [typescript-eslint/typescript-eslint#8735]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/ban-types/

### Description

https://typescript-eslint.io/play/#ts=5.4.2&fileType=.ts&code=C4TwDgpgBANg9gdwgJwPICMBWUC8UDeAUFFAPSlQAWcA5hIQL4DchhoksiKGmATLgWaEgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AIwEMnajTUMkRNGgB7aJHBgAviGlA&tsconfig=&tokens=false

I believe this behavior might be unintended, as the presence of comments should ideally not affect the linting rules for identifying empty object types. It would be beneficial for code quality and consistency if the ban-types rule could be enhanced to trigger an error regardless of the presence of comments within the type object.

I propose an enhancement to the ban-types rule to ensure it consistently identifies and flags empty object types, even when comments are present inside the type declaration. This change would help developers maintain higher code standards and avoid potential issues related to the use of empty object types in TypeScript.

Thank you for considering this enhancement to the ban-types rule. I believe it will make a valuable contribution to the robustness and consistency of TypeScript linting with ESLint.

### Fail

```TypeScript
type lowerObj = {
  // Expecting a lint error"
};
```

### Pass

```TypeScript
type lowerObj2 = {};
```

**Comment by bradzacher** [comment]
This is expected - the rule just matches on raw strings.
It's simplistic by design - it relies on a very fast heuristic to match banned names - specifically it removes the spaces from the found type name and the configured type name and it compares the two.

At this point in time we're not looking to do anything more complex than that.
Additionally we're considering splitting `{}` out into its own rule (https://github.com/typescript-eslint/typescript-eslint/issues/8700).
So I'm going to close this as working as intended.
